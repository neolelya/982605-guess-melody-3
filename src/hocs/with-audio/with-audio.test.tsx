import * as React from 'react';
import * as renderer from 'react-test-renderer';
import withAudio from './with-audio';

export const noop = () => {
  //  do nothing
};

interface MockComponentProps {
  children: React.ReactNode;
}

const MockComponent = (props: MockComponentProps) => {
  const {children} = props;

  return <div>{children}</div>;
};

const MockComponentWrapped = withAudio(MockComponent);

it(`Should withAudio render correctly`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped
          isPlaying={false}
          onPlayButtonClick={noop}
          src={``}
        />,
        {
          createNodeMock() {
            return {};
          },
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
