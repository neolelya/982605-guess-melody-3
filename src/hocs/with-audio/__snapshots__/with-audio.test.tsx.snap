// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should withAudio render correctly 1`] = `
<div>
  <audio />
</div>
`;
